lang/en/docs/cli/actions/add-software.md

```diff
@@ -1,34 +1,238 @@
+Users can compile their own software on Mat3ra cluster via the
+[Command Line Interface](../overview.md) (CLI). This is helpful, for example,
+after introducing some changes or patches to the source code, or the user need
+to run a specific version of an application that is not installed in Mat3ra
+clusters. Most of Mat3ra applications are currently distributed as
+Apptainer[^1] (Singularity[^2]) containers, bundled with all required
+dependencies. This ensures that each application is isolated and avoids
+dependency conflicts. If you plan to run an application that is not installed in
+our cluster, we encourage you to package your code and its dependencies as an
+Apptainer/<wbr/>Singularity container. If you already have a Docker image, it
+can be converted into an Apptainer/<wbr/>Singularity image.
 
-We also explain how to add python packages to the environment [in this page](create-python-env.md).
+## Experiment in the Sandbox mode
 
-## Example: New Quantum ESPRESSO Version
+Apptainer sandbox mode is helpful for testing and fine-tuning the build steps
+interactively. Initialize a sandbox with `--sandbox` or `-s` flag:
+```bash
+apptainer build --sandbox qe_sandbox/ docker://almalinux:9
+```
+
+The above command will extract the entire Linux OS tree (`/bin`, `/etc`, `/usr`)
+from AlmaLinux 9 Docker image to a standard directory named `qe_sandbox`.
+
+Now to install packages and save them to the sandbox folder, we can enter into
+the container in shell (interactive) mode with write permission (use
+`--writable` or `-w` flag). We will also need `--fakeroot` or `-f` flag to
+install software as root inside the container:
+```bash
+apptainer shell --writable --fakeroot qe_sandbox/
+```
+
+Once, inside the Apptainer shell, we can install packages and run commands
+interactively, as we would normally do from the terminal, for example:
+```bash
+dnf install gcc
+```
+
+Once you are happy with the sandbox, tested the build steps, and installed
+everything you need, `exit` from the Apptainer shell mode.
+
+
+## Build container
+
+We may either package the sandbox directory into a final image:
+```bash
+apptainer build -f espresso.sif qe_sandbox/
+```
+
+After the container is build and saved as SIF image, we may delete our sandbox
+folder. We need to set appropriate permission in order to be able to delete:
+```bash
+chmod -R u+rwX qe_sandbox
+rm -rf qe_sandbox
+```
+
+Alternative to converting the sandbox folder to SIF image, we may create an
+Apptainer definition with the finalized build steps. Below is an example
+Apptainer/<wbr/>Singularity definition to build Quantum ESPRESSO container
+along with its dependencies.
+
```

Comment:
This file is pretty long, let's put into `expand` block?

Reply:
@svbarabash to see how `expand` is implemented https://docs-new.mat3ra.com/cli/actions/add-software/?h=add+new#build-from-a-definition-file